pytorch_lightning/utilities/arguments.py

```diff
@@ -0,0 +1,127 @@
+import inspect
+from argparse import ArgumentParser, Namespace
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import pytorch_lightning as pl
+
+
+@dataclass(frozen=True)
+class LitArg:
+
+    name: str
+    types: tuple
+    default: Any
+    required: bool = False
+    context: Optional[str] = None
+
+
+class LightningArgumentParser(ArgumentParser):
+    def __init__(self, *args, ignore_required_init_args=True, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ignore_required_init_args = ignore_required_init_args
```

Comment:
perhaps move this to `add_object_args` to both keep the init in line w/ its parent and allow more fine grained control over how we handle each object's required init args.